test/test_matplotlib.py

```diff
@@ -28,3 +28,17 @@ def test_pdf(selenium):
     selenium.run("import io")
     selenium.run("fd = io.BytesIO()")
     selenium.run("plt.savefig(fd, format='pdf')")
+
+def test_rendering(selenium):
+    selenium.load_package("matplotlib")
+    selenium.run("from matplotlib import pyplot as plt")
+    selenium.run("import numpy as np")
+    selenium.run("t = np.arange(0.0, 2.0, 0.01)")
+    selenium.run("s = 1 + np.sin(2 * np.pi * t)")
+    selenium.run("plt.plot(t, s, linewidth=1.0, marker=11)")
+    selenium.run("plt.plot(t, t)")
+    selenium.run("plt.grid(True)")
+    selenium.run("plt.show()")
+    canvas_data = selenium.run("plt.gcf().canvas.get_pixel_data()")
+    assert canvas == '1'
```

Comment:
A couple thoughts:

1. You could use a multi-line string for this (the tests above in this file predate support for that -- maybe you could update those as well while you're here).

2. You probably don't want or need to return the data over to the pytest side here.  You could just do all the comparison work within Pyodide and return a boolean and assert that that boolean is True.

Reply:
About 2nd, If I don't return data to the pytest side i.e. if I use 
`selenium.run("canvas_data = plt.gcf().canvas.get_pixel_data()")`

How will I verify the data of the reference image which will probably be read from a folder. This is because reading the data of the reference image would always be on the `pytest` side. How can the reference data be put inside the `pyodide` side too?

Reply:
You could fetch the data from the pytest side to the pyodide side.  Files in `test/data` are available from the testing webserver.  The reason I suggest doing it this way is that sending the data back to selenium causes a conversion to Javascript and then to Python again that may be hard to deal with.